lib/kennel/id_map.rb

```diff
@@ -0,0 +1,26 @@
+# frozen_string_literal: true
+module Kennel
+  class IdMap
+    NEW = :new
+
+    def initialize
+      @map = Hash.new { |h, k| h[k] = {} }
+    end
+
+    def add(type, tracking_id, id)
+      @map[type][tracking_id] = id
+    end
+
+    def add_new(type, tracking_id)
+      @map[type][tracking_id] = NEW
+    end
```

Comment:
not a big fan of this interface since it would require 2 new methods for each thing (like NEW), better to set `IdMap::NEW` and compare against `IdMap::NEW` ?